staging/src/k8s.io/pod-security-admission/policy/check_allowPrivilegeEscalation.go

```diff
@@ -50,44 +49,51 @@ func CheckAllowPrivilegeEscalation() Check {
 				// Field added in 1.8:
 				// https://github.com/kubernetes/kubernetes/blob/v1.8.0/staging/src/k8s.io/api/core/v1/types.go#L4797-L4804
 				MinimumVersion: api.MajorMinorVersion(1, 8),
-				CheckPod:       allowPrivilegeEscalation_1_8,
+				CheckPod:       withOptions(allowPrivilegeEscalation_1_8),
 			},
 			{
 				// Starting 1.25, windows pods would be exempted from this check using pod.spec.os field when set to windows.
 				MinimumVersion: api.MajorMinorVersion(1, 25),
-				CheckPod:       allowPrivilegeEscalation_1_25,
+				CheckPod:       withOptions(allowPrivilegeEscalation_1_25),
 			},
 		},
 	}
 }
 
-func allowPrivilegeEscalation_1_8(podMetadata *metav1.ObjectMeta, podSpec *corev1.PodSpec) CheckResult {
-	var badContainers []string
-	visitContainers(podSpec, func(container *corev1.Container) {
-		if container.SecurityContext == nil || container.SecurityContext.AllowPrivilegeEscalation == nil || *container.SecurityContext.AllowPrivilegeEscalation {
-			badContainers = append(badContainers, container.Name)
+func allowPrivilegeEscalation_1_8(podMetadata *metav1.ObjectMeta, podSpec *corev1.PodSpec, opts options) CheckResult {
+	var badContainers violations[string]
+
+	visitContainersWithPath(podSpec, func(container *corev1.Container, pathFn PathFn) {
+		pathFn = pathFn.child("securityContext").child("allowPrivilegeEscalation")
+		if container.SecurityContext == nil {
+			badContainers.Add(container.Name, opts, required(pathFn))
```

Comment:
Wrong path.

Reply:
> For example, as reviewed here, when container.SecurityContext == nil, the returned Field should be set to spec.containers[*].securityContext.allowPrivilegeEscalation or spec.containers[*].securityContext?

I would indicate the specific field needed, e.g.: `spec.containers[*].securityContext.allowPrivilegeEscalation`

> when container did not explicitly set seccompProfile and also no valid pod-level seccompProfile, what value should we assign to the Field? spec.securityContext.seccompProfile.type or spec.containers[*].securityContext.seccompProfile.type?

since we'll be returning a single error, I would use the pod-level fieldPath

Reply:
> since we'll be returning a single error, I would use the pod-level fieldPath

@liggitt  If we just return a single error which using the pod-level fieldPath, this does not seem to give the user enough information.

For example, when there have no valid pod-level seccompProfile, and there also have multiple container that not explicitly set seccompProfile. In this scenario, if we just return a single error which using the pod-level fieldPath, users are unable to determine which specific containers are causing errors based on the returned field error list.

we think this information is still necessary, and if we have it, we can exempt specific container in external system.